src/views/Users/ViewUser/index.jsx

```diff
@@ -0,0 +1,394 @@
+        return entry;
+      }
+
+      const result = entry;
+
+      result.metadata[key] = value;
+
+      return result;
+    };
+
+    return permission.metadata.isAdditional
+      ? setAdditionalPermissions(additionalPermissions.map(updateText))
+      : setPermissions(permissions.map(updateText));
+  };
+
+  const handleUserSave = () => {};
+  const handleUserDelete = () => {};
+  const renderRole = (role, index) => (
+    <Grid container spacing={2} key={index}>
+      <Grid item xs={11}>
+        <TextField
+          disabled={role.metadata.isAdditional ? false : !isNewUser}
+          onChange={e => handleRoleNameChange(role, index, e.target.value)}
+          value={role.name}
+          fullWidth
+        />
+      </Grid>
+      <Grid item xs={1} className={classes.gridDelete}>
+        <IconButton onClick={() => handleRoleDelete(role, index)}>
+          <DeleteIcon />
+        </IconButton>
+      </Grid>
+    </Grid>
+  );
+  const renderPermission = (permission, index) => (
+    <Grid container spacing={2} key={index}>
+      <Grid item xs={3}>
+        <AutoCompleteText
+          value={defaultToEmptyString(permission.name)}
+          onValueChange={handlePermissionNameChange(permission, index)}
+          getSuggestions={getSuggestions(allPermissions)}
+          label="Name"
+          required
+          disabled={!permission.metadata.isAdditional}
+          inputProps={{
+            autoFocus: true,
+          }}
+        />
+      </Grid>
+      <Grid item xs={4}>
+        <AutoCompleteText
+          multi
+          disabled={!supportsProductRestriction(permission.name)}
+          selectedItems={permission.options.products}
+          onSelectedItemsChange={handleRestrictionChange(
+            permission,
+            'products'
+          )}
+          onValueChange={handleRestrictionTextChange(permission, 'productText')}
+          value={permission.metadata.productText}
+          getSuggestions={getSuggestions(products)}
+          label="Product Restrictions"
+          inputProps={{
+            autoFocus: true,
+          }}
+        />
+      </Grid>
+      <Grid item xs={4}>
+        <AutoCompleteText
+          multi
+          disabled={!supportsActionRestriction(permission.name)}
+          selectedItems={permission.options.actions}
+          onSelectedItemsChange={handleRestrictionChange(permission, 'actions')}
+          onValueChange={handleRestrictionTextChange(permission, 'actionText')}
+          value={permission.metadata.actionText}
+          getSuggestions={getSuggestions(getSupportedActions(permission.name))}
+          label="Action Restrictions"
+          inputProps={{
+            autoFocus: true,
+          }}
+        />
+      </Grid>
+      <Grid item xs={1} className={classes.gridDelete}>
+        <IconButton onClick={() => handlePermissionDelete(permission, index)}>
+          <DeleteIcon />
+        </IconButton>
+      </Grid>
+    </Grid>
+  );
```

Comment:
Clicking on the plus button allows a user to add a permission but product restrictions and action restrictions are disabled for that row. Is this expected?

![Screen Shot 2019-07-15 at 11 12 42 AM](https://user-images.githubusercontent.com/3766511/61227163-cbeac280-a6f1-11e9-9e99-c9f9054db746.png)

Reply:
Yup. They'll become enabled (where appropriate) when you enter a valid permission name. For example, the `rule` permission supports both types of restrictions, but `admin` only supports product restrictions.